test/presets/aws-lambda.test.ts

```diff
@@ -62,21 +57,23 @@ describe("nitro:preset:aws-lambda", async () => {
         body: body || "",
       };
       const res = await handler(event);
-      const resHeaders = { ...res.headers };
-      if (res.cookies) {
-        if (!resHeaders["set-cookie"]) {
-          resHeaders["set-cookie"] = [];
-        }
-        if (!Array.isArray(resHeaders["set-cookie"])) {
-          resHeaders["set-cookie"] = [resHeaders["set-cookie"]];
-        }
-        resHeaders["set-cookie"].push(...res.cookies);
-      }
-      return {
-        data: destr(res.body),
-        status: res.statusCode,
-        headers: resHeaders,
-      };
+      return makeResponse(res, "v2");
     };
   });
 });
+
+const makeResponse = (response: any, version: "v1" | "v2") => {
+  const headers = { ...response.headers };
```

Comment:
Perf: We don't need to duplicate memory for headers if there is no `set-cookie` in it (nor the version checks logic). I wouldn't also make a copy even for `set-cookie` unless it is necessary to prevent a valid scenario where modifying original headers makes issues